master/buildbot/steps/source/git.py

```diff
@@ -697,3 +697,78 @@ def _doPush(self):
 
         ret = yield self._dovccmd(cmd)
         return ret
+
+
+class GitTag(buildstep.BuildStep, GitStepMixin, CompositeStepMixin):
+
+    description = None
+    descriptionDone = None
+    descriptionSuffix = None
+
+    name = 'gittag'
+    renderables = ['repourl', 'name', 'messages']
+
+    def __init__(self, workdir=None, tagName=None,
+                 annotated=False, messages=None, force=False, env=None,
+                 timeout=20 * 60, logEnviron=True, config=None, **kwargs):
+
+        self.workdir = workdir
+        self.tagName = tagName
+        self.annotated = annotated
+        self.messages = messages
+        self.force = force
+        self.env = env
+        self.timeout = timeout
+        self.logEnviron = logEnviron
+        self.config = config
+
+        # These attributes are required for GitStepMixin but not useful to tag
+        self.repourl = " "
+        self.sshHostKey = None
+        self.sshPrivateKey = None
+
+        super().__init__(**kwargs)
+
+        self.setupGitStep()
+
+        if not self.tagName:
+            bbconfig.error('GitTag: must provide tagName')
+
+        if self.annotated and not self.messages:
+            bbconfig.error('GitTag: must provide messages in case of annotated tag')
+
+        if not self.annotated and self.messages:
+            bbconfig.error('GitTag: messages are required only in case of annotated tag')
+
+        if self.messages and not isinstance(self.messages, list):
+            bbconfig.error('GitTag: messages should be a list')
+
+    @defer.inlineCallbacks
+    def run(self):
+        self.stdio_log = yield self.addLog("stdio")
+        gitInstalled = yield self.checkTagSupport()
+
+        if not gitInstalled:
```

Comment:
Coverage says this branch is not covered. Could we have a small unit test for this condition?

Reply:
Yeah I spotted it, I will do it and add some more case to current unit test to cover identical branch in other buildstep.